### DISCUSSION TITLE
Issues running MOOSE on HPC

### DISCUSSION THREAD

**Discussion by TheGreatCid** [idaholab/moose#20316] [Q&A General] [answered]
Hi,

I have compiled Petsc and Libmesh with `./update_and_rebuild_libmesh` and `update_and_rebuildt_petsc` with the following modules loaded
 **cmake/3.19.3-rhel8
Python/3.8.1
GCC/9.3.0
MPICH/3.2.1**

MOOSE then compiles without issue.

However, when I try to run a simulation, I am greeted with
 `[6] ../src/utils/xdr_cxx.C, line 148, compiled Feb 15 2022 at 23:27:32
ERROR: Functionality is not available.
Make sure LIBMESH_HAVE_XDR is defined at build time
The XDR interface is not available in this installation`

I am on the most recent commit of MOOSE and I have no issues running simulations locally. 
Is anyone familiar with this error?

**Comment by GiudGiud** [comment]
Hello

Are you trying to use the restart functionality or is this coming up out of the blue?
Locally did you install with the conda/mamba build or still with `update_and_rebuild_libmesh` ?
There should be a log of the libmesh install, in the scripts directory iirc, could you please share it?

Guillaume

**Reply by TheGreatCid** [reply]
Hi, 
Thank you for your response.
Locally I installed with mamba build
[libmesh_diagnostic.log](https://github.com/idaholab/moose/files/8089744/libmesh_diagnostic.log)

I cannot find the full build log, but I did find the configure log.

In the script file, it should be outputting a build log in the scripts directory, but for some reason that did not happen

**Reply by TheGreatCid** [reply]
Ah, but I do have a combination of the petsc and libmesh build log from a build I did the day before. It should be identical except for the fact that this was built with a slightly older version of GCC. 

[make_raccoon.txt](https://github.com/idaholab/moose/files/8089842/make_raccoon.txt)

**Comment by milljm** [comment, marked as answer]
```pre
LIBMESH_HAVE_XDR
```
Comes from the system not having libtirpc installed. Basically the rpc.h files specifically. This as a reference: https://bugzilla.redhat.com/show_bug.cgi?id=1537432

**Reply by TheGreatCid** [reply]
Thank you!

**Reply by milljm** [reply]
I hope this is the solution! Please let us know either way 😃. We install the following on our build machines:
```pre
libtirpc libtirpc-devel
```

**Reply by TheGreatCid** [reply]
Hi!
I went ahead and installed the conda/mamba environment on my HPC and that fixed my issue, so I did not apply this solution. 
I'll go ahead and mark it as the answer in case it helps anyone else!
Thanks!
